File: UNIDAD 1/PROPAGACION DE ERRORES/TEOREMA DE BOLZANO/Bolzano.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Main.java to edit this template
 */
package com.mycompany.mavenproject3;

/**
 *
 * @author Adrian Nole PC
 */
public class Bolzano {

    public static double function(double x) {
        // Define la función para la cual deseas encontrar el cero
        return x * x - 4;
    }

    public static void main(String[] args) {
        double a = 1; // Extremo izquierdo del intervalo
        double b = 3; // Extremo derecho del intervalo
        double epsilon = 0.0001; // Tolerancia para la aproximación del cero

        if (function(a) * function(b) > 0) {
            System.out.println("No se puede aplicar el teorema de Bolzano en el intervalo [" + a + ", " + b + "]");
            return;
        }

        double c;
        do {
            c = (a + b) / 2; // Calcula el punto medio del intervalo
            if (function(c) == 0) {
                break; // Se encontró el cero exacto
            } else if (function(a) * function(c) < 0) {
                b = c; // El cero se encuentra en el intervalo [a, c]
            } else {
                a = c; // El cero se encuentra en el intervalo [c, b]
            }
        } while (Math.abs(function(c)) > epsilon);

        System.out.println("El cero aproximado es: " + c);
    }
}

File: UNIDAD 1/Proyecto_Numericos/Bolzano.js
function functionBolzano(x) {
    // Define la función para la cual deseas encontrar el cero
    return x * x - 4;
  }
  
  function encontrarCeroBolzano(a, b, epsilon) {
    if (functionBolzano(a) * functionBolzano(b) > 0) {
      return null;
    }
  
    let c;
    do {
      c = (a + b) / 2; // Calcula el punto medio del intervalo
      if (functionBolzano(c) === 0) {
        break; // Se encontró el cero exacto
      } else if (functionBolzano(a) * functionBolzano(c) < 0) {
        b = c; // El cero se encuentra en el intervalo [a, c]
      } else {
        a = c; // El cero se encuentra en el intervalo [c, b]
      }
    } while (Math.abs(functionBolzano(c)) > epsilon);
  
    return c;
  }
  
  function ejecutarProgramaBolzano() {
    const a = parseFloat(prompt("Ingrese el extremo izquierdo del intervalo:"));
    const b = parseFloat(prompt("Ingrese el extremo derecho del intervalo:"));
    const epsilon = parseFloat(prompt("Ingrese la tolerancia para la aproximación del cero:"));
  
    const ceroAproximado = encontrarCeroBolzano(a, b, epsilon);
    const resultadoElement = document.getElementById('resultado');
    if (ceroAproximado !== null) {
      resultadoElement.textContent = "El cero aproximado es: " + ceroAproximado;
    } else {
      resultadoElement.textContent = "No se puede aplicar el teorema de Bolzano en el intervalo [" + a + ", " + b + "]";
    }
  }